presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -553,6 +562,15 @@ public Signature resolveFunction(QualifiedName name, List<TypeSignature> paramet
             return Optional.of(mostSpecificFunction.getBoundSignature());
         }
 
+        // Prefer functions where all arguments are of the same type
```

Comment:
This is to deal with things like MULTIPLY where you have MULTIPLY(bigint,bigint) and MULTIPLY(bigint, inteval_year_to_month)  and MULTIPLY(bigint, interval_year_to_day) and other similar cases.   Here we say to always choose MULTIPLY(bigint,bigint). I could instead register specific functions for MULTIPLY(bigint, unknown) and do the same for a slew of other similar cases.  This was more concise, and maintainable going forward, since you don't need to then register more unknown functions if for example something similar were added for decimal.